ecl/resource_ecl_baremetal_server_v2.go

```diff
@@ -0,0 +1,593 @@
+						},
+					},
+				},
+			},
+			"personality": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"path": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"contents": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceBaremetalServerV2Create(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	baremetalClient, err := config.baremetalV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating ECL baremetal client: %s", err)
+	}
+
+	var createOpts servers.CreateOptsBuilder
+
+	// Determines the Image ID using the following rules:
+	// If a bootable block_device was specified, ignore the image altogether.
+	// If an image_id was specified, use it.
+	// If an image_name was specified, look up the image ID, report if error.
+	imageID, err := getBaremetalImageID(baremetalClient, d)
+	if err != nil {
+		return err
+	}
+
+	flavorID, err := getBaremetalFlavorID(baremetalClient, d)
+	if err != nil {
+		return err
+	}
+
+	networks, err := getBaremetalNetworks(baremetalClient, d)
+	if err != nil {
+		return err
+	}
+
+	raidArrays, err := getBaremetalRaidArrays(baremetalClient, d)
+	if err != nil {
+		return err
+	}
+
+	lvmVolumeGroups, err := getBaremetalLVMVolumeGroups(baremetalClient, d)
+	if err != nil {
+		return err
+	}
+
+	filesystems, err := getBaremetalFilesystems(baremetalClient, d)
+	if err != nil {
+		return err
+	}
+
+	personality, err := getBaremetalPersonalities(baremetalClient, d)
+	if err != nil {
+		return err
+	}
+
+	createOpts = &servers.CreateOpts{
+		Name:             d.Get("name").(string),
+		AdminPass:        d.Get("admin_pass").(string),
+		KeyName:          d.Get("key_pair").(string),
+		ImageRef:         imageID,
+		FlavorRef:        flavorID,
+		AvailabilityZone: d.Get("availability_zone").(string),
+		Networks:         networks,
+		Metadata:         resourceInstanceMetadataV2(d),
+		UserData:         []byte(d.Get("user_data").(string)),
+		RaidArrays:       raidArrays,
+		LVMVolumeGroups:  lvmVolumeGroups,
+		Filesystems:      filesystems,
+		Personality:      personality,
+	}
+
+	log.Printf("[DEBUG] Create Options: %#v", createOpts)
+
+	var server *servers.Server
+	server, err = servers.Create(baremetalClient, createOpts).Extract()
+	if err != nil {
+		return fmt.Errorf("Error creating baremetal server: %s", err)
+	}
+
+	log.Printf("[INFO] baremetal server ID: %s", server.ID)
+
+	log.Printf("[DEBUG] Waiting for baremetal server (%s) to become available", server.ID)
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"BUILD"},
+		Target:     []string{"ACTIVE"},
+		Refresh:    waitForBaremetalServerActive(baremetalClient, server.ID),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      5 * time.Second,
```

Comment:
Bare-metal server takes relatively long time to complete each CUD steps, so I think it is better to define explicit PollInterval for this resource.
Ex. In Common Function Gateway, it will take over 30 minutes to complete creation, so interval is defined as 1 minutes.

Reply:
got it!